iOS/samples/sdksample/BingMapsSdkSample/ViewController.swift

```diff
@@ -107,6 +109,24 @@ class ViewController: UIViewController, UIPickerViewDelegate, UIPickerViewDataSo
         }
 
         setupDemoMenu()
+        updateMapViewColorScheme()
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        if (self.traitCollection.userInterfaceStyle != previousTraitCollection?.userInterfaceStyle) {
+            updateMapViewColorScheme()
+        }
+    }
+
+    func updateMapViewColorScheme() {
+        let index = mapStyleStrings.firstIndex(of:
+            self.traitCollection.userInterfaceStyle == .dark ?
```

Comment:
For completeness and correctness, we should also factor the current style so the app updates between high contrast dark <-> high contrast light, road dark <-> road light, but exclude changes for all others.
We can also enable canvas dark <-> canvas light, since the canvas dark is public but need to create a style sheet using the JSON API for that.

Reply:
Good point on supporting compatible pairs!
As for canvas dark, let's enable it when it's supported within the SDK.